widget/entry.go

```diff
@@ -41,6 +42,10 @@ type Entry struct {
 	MultiLine   bool
 	Wrapping    fyne.TextWrap
 
+	RegexValidation *regexp.Regexp
```

Comment:
I just thought - RegexValidation also implies there could be multiple types of validation (which I agree with) - but why bake this specific one into entry? Perhaps "Validation" can be provided by a "RegexpValidation" provider?